lib/internal/bootstrap_node.js

```diff
@@ -268,15 +268,24 @@
     var console;
     Object.defineProperty(global, 'console', {
       configurable: true,
-      enumerable: true,
+      enumerable: false,
       get: function() {
         if (!console) {
           console = NativeModule.require('console');
           installInspectorConsoleIfNeeded(console,
                                           inspectorConsole,
                                           wrapConsoleCall);
         }
+        Object.defineProperty(global, 'console', {
+          configurable: true,
+          writable: true,
+          enumerable: false,
+          value: console
+        });
         return console;
+      },
+      set: function(customConsole) {
+        console = customConsole;
```

Comment:
This assignment will invalidate the `if (console)` check above in the getter, in the case that `global.console` is deliberately set to any falsy value like `global.console = undefined;`. Instead, do the same thing you did in the getter and use `Object.defineProperty()` on `global`.